src/components/Map/algorithms/Chart.tsx

```diff
@@ -0,0 +1,135 @@
+  '5': 
+    {"w": {"u": 0.11447557860207201, "v": 0.5253881911125526, "w": 7.548832973261137}, 
+    "u": {"u": 1.1971055926702734, "v": 0.3869554744587899, "w": 0.5306180750216268}, 
+    "v": {"u": 0.08736238508669167, "v": 5.302357097157161, "w": 0.5498104456545143}}, 
+  '3': 
+    {"u": {"w": 0.5846156302145143, "u": 0.7100239287446046, "v": 0.3458195569108133}, 
+    "w": {"u": 0.12356102280699693, "v": 0.5789979029092013, "w": 3.3594015098012524}, 
+    "v": {"v": 3.15260229690905, "u": 0.07788491452768159, "w": 0.6169780536897971}}, 
+  '4': 
+    {"w": {"v": 0.4474437258633965, "u": 0.10132558606505494, "w": 4.382201150288006}, 
+    "u": {"u": 0.9657539481298845, "v": 0.44933065647196085, "w": 0.459774563353038}, 
+    "v": {"u": 0.10096360051946618, "v": 4.298012880591784, "w": 0.45620812795185256}}, 
+  '2': 
+    {"v": {"w": 0.4560563664334602, "u": 0.1013954341203142, "v": 2.1497846458869967}, 
+    "u": {"w": 0.46385720089360516, "v": 0.4551294396765634, "u": 0.4789370417123615}, 
+    "w": {"v": 0.4474339020518443, "w": 2.191212980781382, "u": 0.10138603482247471}}
+  } 
+  const labels = Object.keys(diffcen);
+
+  const node_to_time_values = new Map<string, number[]>();
+
+  for (const [time, time_map] of Object.entries(diffcen)) {
+    for (const [source_node, node_map] of Object.entries(time_map)) {
+      for (const [destination_node, value] of Object.entries(node_map)) {
+        if (!node_to_time_values.has(source_node)) {
+          node_to_time_values.set(source_node, []);
+        }
+        if (source_node === destination_node) {
+          node_to_time_values.get(source_node)?.push(value);
+        }
+      }
+    }
+  }
+  
+
+  const datasets: { fill: boolean; label: string; data: number[]; borderColor: string; backgroundColor: string; }[] = [];
+  const colors = ["53B3AD", "4146C2", "E78B37", "CD4B81", "7F83F3", "8EDE78", "3878EB", "6A2BCB", "E2C742", "BE6325"]
+
+  // sort by value and assign color from left to right
+  const sorted = new Map([...node_to_time_values.entries()].sort((a, b) => b[1][0] - a[1][0]));
+  let i = 0;
+  sorted.forEach((values, node_name) => {
+    const color_rgb = hexToRgb(colors[i]);
+    if (color_rgb) {
+      colors[i] = `rgba(${color_rgb.r}, ${color_rgb.g}, ${color_rgb.b}, 0.9)`;
+    }
+
+    datasets.push({
+      fill: true,
+      label: node_name,
+      data: values,
+      borderColor: colors[i],
+      backgroundColor: colors[i],
+    });
+    i++;
+  });
+
+  // reverse the order of the datasets so that the legend is in the same order as the graph
+  datasets.reverse();
+  
+
+  return {
+    labels,
+    datasets: datasets,
+  };
+}
+
+// Credit: https://stackoverflow.com/questions/5623838/rgb-to-hex-and-hex-to-rgb
+function hexToRgb(hex: string): { r: number; g: number; b: number } | null {
+  // Expand shorthand form (e.g. "03F") to full form (e.g. "0033FF")
+  const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i;
```

Comment:
I'm not super keen on regex without explanation because it's so obfuscated. That said, [here's](https://stackoverflow.com/questions/36686461/why-does-regex-not-match-when-there-are-more-then-one) an explanation of the same GitHub code and a further simplification.